02week/ticTacToe.js

```diff
@@ -24,23 +26,65 @@ function printBoard() {
 }
 
 function horizontalWin() {
-  // Your code here
+  if ((board[0][0] === 'X') && (board[0][1] === 'X') && (board[0][2] === 'X')) {
+    return true;
+  } else if ((board[1][0] === 'X') && (board[1][1] === 'X') && (board[1][2] === 'X')) {
+    return true;
+  } else if ((board[2][0] === 'X') && (board[2][1] === 'X') && (board[2][2] === 'X')) {
+    return true;
+  } else if ((board[0][0] === 'O') && (board[0][1] === 'O') && (board[0][2] === 'O')) {
+    return true;
+  } else if ((board[1][0] === 'O') && (board[1][1] === 'O') && (board[1][2] === 'O')) {
+    return true;
+  } else if ((board[2][0] === 'O') && (board[2][1] === 'O') && (board[2][2] === 'O')) {
+    return true;
+  }
 }
 
+
 function verticalWin() {
-  // Your code here
+  if ((board[0][0] === 'X') && (board[1][0] === 'X') && (board[2][0] === 'X')) {
+    return true;
+  } else if ((board[0][1] === 'X') && (board[1][1] === 'X') && (board[2][1] === 'X')) {
+    return true;
+  } else if ((board[0][2] === 'X') && (board[1][2] === 'X') && (board[2][2] === 'X')) {
+    return true;
+  } else if ((board[0][0] === 'O') && (board[1][0] === 'O') && (board[2][0] === 'O')) {
+    return true;
+  } else if ((board[0][1] === 'O') && (board[1][1] === 'O') && (board[2][1] === 'O')) {
+    return true;
+  } else if ((board[0][2] === 'O') && (board[1][2] === 'O') && (board[2][2] === 'O')) {
+    return true;
+  }
 }
 
 function diagonalWin() {
-  // Your code here
+  if ((board[0][0] === 'X') && (board[1][1] === 'X') && (board[2][2] === 'X')) {
+    return true;
+  } else if ((board[0][2] === 'X') && (board[1][1] === 'X') && (board[2][0] === 'X')) {
+    return true;
+  } else if ((board[0][0] === 'O') && (board[1][1] === 'O') && (board[2][2] === 'O')) {
+    return true;
+  } else if ((board[0][2] === 'O') && (board[1][1] === 'O') && (board[2][2] === 'O')) {
+    return true;
+  }
 }
 
 function checkForWin() {
-  // Your code here
+  horizontalWin();
+  verticalWin();
+  diagonalWin();
+  return true;
 }
 
 function ticTacToe(row, column) {
-  // Your code here
+  board[row][column] = playerTurn;
+  if (playerTurn === 'X') {
+    (playerTurn = 'O');
+  } else {
+    (playerTurn = 'X');
+  }
+  checkForWin();
```

Comment:
You should only switch the turn if there is no win

Reply:
Fixed.